AnkiDroid/build.gradle

```
@@ -207,6 +207,24 @@ task installGitHook(type: Copy) {
 }
 tasks.getByPath(':AnkiDroid:preBuild').dependsOn installGitHook
 
+// Issue 11078 - some emulators run, but run zero tests, and still report success
+task assertNonzeroAndroidTests() {
+    doLast {
+        File folder = file("./build/outputs/androidTest-results/connected/flavors/play")
+        File[] listOfFiles = folder.listFiles({d, f-> f ==~ /.*.xml/ } as FilenameFilter)
+        for (File file : listOfFiles) {
+            String[] matches = file.readLines().findAll { it.contains('testsuite tests="0"')}
```

We should be testing for the presence of a known good (nonzero) value, rather than the absence of a value. 

This one would fail silently if the format of the string changed

---

Hmm I don't want to be in the business of having the test memorize the correct number of tests but you are right that this will fail silently

How about I positively match on the regex equivalent of `\d+` in the tests match to make sure I am matching a test count, then I extract the number and make sure it is not zero?

That way if format changes I will fail to match at all, but successful match of bad number (0) will fail too?

---

That's my thoughts: we don't want the exact number, but: 

* we want there to be **A** number
* we don't want that number to be 0

---

That's my thoughts: we don't want the exact number, but: 

* we want there to be **A** number
* we don't want that number to be 0

---

working on this, will re-ping when done - other items are done but not pushed

---

fixed!